lib/matplotlib/collections.py

```diff
@@ -1254,6 +1255,285 @@ def set_verts_and_codes(self, verts, codes):
+        ----------
+        t_direction : {'x', 'y'}
+            The axes on which the variable lies.
+
+            - 'x': the curves are ``(t, f1)`` and ``(t, f2)``.
+            - 'y': the curves are ``(f1, t)`` and ``(f2, t)``.
+
+        t : array (length N)
+            The ``t_direction`` coordinates of the nodes defining the curves.
+
+        f1 : array (length N) or scalar
+            The other coordinates of the nodes defining the first curve.
+
+        f2 : array (length N) or scalar
+            The other coordinates of the nodes defining the second curve.
+
+        where : array of bool (length N), optional
+            Define *where* to exclude some {dir} regions from being filled.
+            The filled regions are defined by the coordinates ``t[where]``.
+            More precisely, fill between ``t[i]`` and ``t[i+1]`` if
+            ``where[i] and where[i+1]``.  Note that this definition implies
+            that an isolated *True* value between two *False* values in *where*
+            will not result in filling.  Both sides of the *True* position
+            remain unfilled due to the adjacent *False* values.
+
+        interpolate : bool, default: False
+            This option is only relevant if *where* is used and the two curves
+            are crossing each other.
+
+            Semantically, *where* is often used for *f1* > *f2* or
+            similar.  By default, the nodes of the polygon defining the filled
+            region will only be placed at the positions in the *t* array.
+            Such a polygon cannot describe the above semantics close to the
+            intersection.  The t-sections containing the intersection are
+            simply clipped.
+
+            Setting *interpolate* to *True* will calculate the actual
+            intersection point and extend the filled region up to this point.
+
+        step : {{'pre', 'post', 'mid'}}, optional
+            Define *step* if the filling should be a step function,
+            i.e. constant in between *t*.  The value determines where the
+            step will occur:
+
+            - 'pre': The f value is continued constantly to the left from
+              every *t* position, i.e. the interval ``(t[i-1], t[i]]`` has the
+              value ``f[i]``.
+            - 'post': The y value is continued constantly to the right from
+              every *x* position, i.e. the interval ``[t[i], t[i+1])`` has the
+              value ``f[i]``.
+            - 'mid': Steps occur half-way between the *t* positions.
+
+        axes : `~matplotlib.axes.Axes` or None
+            The owning Axes to which the collection belongs, or None if there is
+            no Axes upon instantiation.
+
+        **kwargs
+            Forwarded to `.PolyCollection`.
+
+        See Also
+        --------
+        .Axes.fill_between, .Axes.fill_betweenx
+        """
+        self.t_direction = t_direction
+        kwargs = self._normalise_kwargs(axes, **kwargs)
```

Comment:
This could just be a call to set_data?

Reply:
Hi, in `set_data`,
- We *do not* make use of the second return-value of `self._make_verts_and_pts`
- After calling `self._make_verts_and_pts`, we call `self.set_verts(polys)`, where `set_verts` comes from the super class

In contrast, here in `__init__`,
- We *do* make use of the second return-value of `self._make_verts_and_pts`
- After calling `self._make_verts_and_pts`, we call the initialiser of the super class, then update axes data limits, making use of `pts`.

To conclude, the steps are different in `__init__` and in `set_data`.

Reply:
After 84c5e559a8a6d6c713bb1a5684ec19d7de40d20e, `_normalise_kwargs` is no more.